daemon/stats.go

```diff
@@ -46,6 +47,12 @@ func (daemon *Daemon) ContainerStats(ctx context.Context, prefixOrName string, c
 
 	outStream := config.OutStream
 	if config.Stream {
+		// Used before the Flush(), so we don't set the header twice because of the otel wrapper
+		// see here: https://github.com/moby/moby/issues/47448
+		if rw, ok := outStream.(http.ResponseWriter); ok {
+			rw.WriteHeader(http.StatusOK)
```

Comment:
This is not the right place to handle this.

I'm also not sure how this fix the otel issue since the very first write to the stream would be writing out the response header.

Reply:
> This is not the right place to handle this.
> ...
> It seems like we either to need to manually instrument the streaming responses or provide a smarter writer implementation to the daemon so that the first flush calls WriteHeader explicitly.

Yeah i mostly agree with this, but there's another couple things to consider here:
- `WriteFlusher` is currently exported from our ioutils pkg, and used by a few other projects (https://grep.app/search?q=.NewWriteFlusher%28&filter[lang][0]=Go);
- We'd have to call Write() instead of WriteHeader() during Flush(), because of how the otel wrapper is written.
 
I have opened [another draft pr](https://github.com/moby/moby/pull/47718) yesterday, moving the current `WriteFlusher` impl and making a new internal one in `/internal/writeflusher`  with these changes. 

If we merge that, this PR can be closed as it'd be unnecessary, but I left it here in case we decide on a different strategy and we only want a quick fix for now.